ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/BlankLineBeforeDeclarationRule.kt

```diff
@@ -71,29 +102,53 @@ public class BlankLineBeforeDeclarationRule :
             return
         }
 
-        node
-            .siblings(false)
-            .takeWhile { it.isWhiteSpace() || it.isPartOfComment() }
-            .lastOrNull()
-            ?.let { previous ->
-                when {
-                    !previous.isWhiteSpace() -> previous
-                    !previous.text.startsWith("\n\n") -> node
-                    else -> null
-                }?.let { insertBeforeNode ->
-                    emit(insertBeforeNode.startOffset, "Expected a blank line for this declaration", true)
-                    if (autoCorrect) {
-                        insertBeforeNode.upsertWhitespaceBeforeMe("\n".plus(node.indent()))
-                    }
+        if (node.elementType == PROPERTY && node.treeParent.elementType == WHEN) {
```

Comment:
should we also check `for`, if else blocks `while` or `do while` as local functions can also be defined in those control flows? For example the below code
```
fun test() {
    for (i in 0..10) {
        fun bar() {}
    }
}
```

Reply:
I am not sure what you mean.

The `no-empty-first-line-in-method-block` enforces that below is not accepted:
```
fun test() {

    for (i in 0..10) {
        fun bar() {}
    }
}
```

Or do you mean that a blank line should be enforced before a control flow if it is not the first code element in function? I do not think that is a good idea. For example, it would not make sense to require a blank line before the `while` in example below:
```
internal fun KtLintRuleEngine.ruleProviders(vararg ruleFilters: RuleFilter): Set<RuleProvider> {
    var ruleProviders = initialRuleProviders()
    val ruleFilterIterator = ruleFilters.iterator()
    while (ruleFilterIterator.hasNext()) {
        val ruleFilter = ruleFilterIterator.next()
        ruleProviders = ruleFilter.filter(ruleProviders)
    }
    return ruleProviders
}
```

Reply:
I means 
```
fun test() {
    for (i in 0..10) {
        fun bar() {}
    }
}
```
should be acceptable well formatted code. I thought your logic is giving warning in this case as you were only checking in class, lambda and function

Reply:
I didn't notice the inner `fun bar`. But it is not giving any problems in the current version of the PR. 

```
fun test() {
    for (i in 0..10) {
        fun bar1() {}
        fun bar2() {}
    }
}
```
is formatted as:
```
fun test() {
    for (i in 0..10) {
        fun bar1() {}

        fun bar2() {}
    }
}
```